### PR TITLE
add_gitignore : adding .gitignore file

### PR DESCRIPTION
**Pull request by bghimireamd**
In this PR we add .gitignore file. Currently the ignores are related to documenation.

### CODE REVIEW

**Review by Rmalavally — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+_images/
+_static/
+_templates/
+_toc.yml
+docBin/
+_doxygen/
+_readthedocs/
+ 
+# avoid duplicating contributing.md due to conf.py
+docs/CHANGELOG.md
```
Comment (amberhassaan) [inline .gitignore:16]: why are these lines needed? These could be legit files that we want in the repository.

**Review by amberhassaan — COMMENTED** [review]
One question and rest is good.

**Comment by junliume** [comment]
@bghimireamd @JehandadKhan @atamazov We discussed about `.gitignore` before and previously decided that it should be user specific and not provided by the repo, e.g. `build` does not necessarily reside in the same directory.
What has changed and now requiring us to add the file?

**Comment by atamazov** [comment]
@junliume I do not know. I have my own .gitignore and do not need this.

**Comment by bghimireamd** [comment]
> @bghimireamd @JehandadKhan @atamazov We discussed about `.gitignore` before and previously decided that it should be user specific and not provided by the repo, e.g. `build` does not necessarily reside in the same directory. What has changed and now requiring us to add the file?

Oh! in that case we can place a hold on this PR. Documentation change was introduced by #2587 , which I believe generates some files during build process. To ignore those files this .gitinore was added.

We can have our own local .gitignore.

**Comment by CAHEK7** [comment]
The good practice is not generating anything in the source directory. Ideally it should be read-only.

**Comment by amberhassaan** [comment]
My humble opinion is that it is common for projects to have a `.gitignore` that serves the need. Folks can modify this `.gitignore` locally or if I recall, there's a way to have a `global .gitignore` that users control: https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

**Comment by CAHEK7** [comment]
> My humble opinion is that it is common for projects to have a `.gitignore` that serves the need.

It depends - if you don't generate anything in the build directory - you don't need default project `.gitignore`. No needs - no serve.

Each user makes `.gitignore` for himself based on a particular IDE or specific workflow - putting all possible combinations of IDE's temporary files and directories and something like `build/*` does not look like a good idea.
